pkg/cardinality/request.go

```diff
@@ -183,14 +183,18 @@ func extractSelector(values url.Values) (matchers []*labels.Matcher, err error)
 	}
 
 	// Ensure stable sorting (improves query results cache hit ratio).
-	slices.SortFunc(matchers, func(a, b *labels.Matcher) bool {
-		if a.Name != b.Name {
-			return a.Name < b.Name
+	slices.SortFunc(matchers, func(a, b *labels.Matcher) int {
+		nameCmp := strings.Compare(a.Name, b.Name)
```

Comment:
Never use `strings.Compare()` please. See the doc:
https://cs.opensource.google/go/go/+/refs/tags/go1.21.2:src/strings/compare.go;l=13

Always convert string comparison into a `<`, `==`, `>`.

Reply:
You can keep using `strings.Compare()` in tests. We don't care about performance there.

Reply:
Thanks for the heads up, I wasn't aware. Will fix.

Reply:
@pracucci `strings.Compare` is used in Prometheus code (that should be why I started using it). Would it make sense if I submit a PR to stop Prometheus from using it also?

Reply:
I just submitted a couple of additional comments about usage in vendored Prometheus. I think it's not a blocker to merge this PR, but I would suggest to open an upstream PR to change them (with benchmarks to justify it). If benchmark doesn't show difference, then there's no justification.

Reply:
Fixed.